### PR TITLE
Add inCollection for Digibib hbz VK #2021

### PR DESCRIPTION
**Pull request by TobiasNx**
See #2021

FEX Daten decided they want all CZ publication be findable in the Digibib hbz
 Verbundkatalog. Since these CZ publications are not part of the genuine NZ we add a new inCollection-statement. It includes NZ+CZ resources.

### CODE REVIEW

File: src/main/resources/alma/fix/relatedRessourcesAndLinks.fix [hunk 1]
```diff
@@ -392,6 +392,21 @@ if exists("@inNZ")
...
+
+if any_equal("MBD  .M", "49HBZ_NETWORK")
+  if any_match("POR  .A", ".*")
+    add_field("@inDigibibVK", "true")
+  else
+    add_field("@inDigibibVK", "true")
+  end
+end
+
+if exists("@inDigibibVK")
```
Comment (blackwinter) [inline src/main/resources/alma/fix/relatedRessourcesAndLinks.fix:405]: Why so complicated? You could simply drop this and move the `add_field` instructions up directly below line 380. Or at least get rid of the variable? Or at the _very least_ get rid of the inner conditional?

File: src/main/resources/alma/fix/relatedRessourcesAndLinks.fix [hunk 2]
```diff
@@ -392,6 +392,21 @@ if exists("@inNZ")
...
+  add_field("inCollection[].$append.id", "https://nrw.digibib.net/search/hbzvk/")
+  add_field("inCollection[].$last.label", "Digibib hbz Verbundkatalog")
```
Comment (blackwinter) [inline src/main/resources/alma/fix/relatedRessourcesAndLinks.fix:407]: Typo:

```suggestion
  add_field("inCollection[].$last.label", "DigiBib hbz Verbundkatalog")
```

**Review by blackwinter — CHANGES_REQUESTED** [review]
Add test in `web/test/tests/IndexIntegrationTest.java`?

```java
{ "inCollection.id:\"https\\://nrw.digibib.net/search/hbzvk/\"", /*->*/ 150 },
```

**Review by blackwinter — APPROVED** [review]
Thanks!

I still think a query test would be nice, but it's your decision...

File: web/test/tests/IndexIntegrationTest.java [hunk 3]
```diff
@@ -67,6 +67,7 @@ public static Collection<Object[]> data() {
 			{ "publication.location:Berlin AND publication.startDate:1993", /*->*/ 1 },
 			{ "publication.location:Berlin AND publication.startDate:[1992 TO 2017]", /*->*/ 5 },
 			{ "inCollection.id:\"http\\://lobid.org/organisations/DE-655#\\!\"", /*->*/ 133 },
+            { "inCollection.id:\"https\\://nrw.digibib.net/search/hbzvk/\"", /*->*/ 150 },
```
Comment (blackwinter) [inline web/test/tests/IndexIntegrationTest.java:70]: Wrong indentation. Should be three tabs (`\t`)?

Comment (TobiasNx) [reply web/test/tests/IndexIntegrationTest.java]: I thought we changed it to spaces. I am not sure why we still have all these tabs as indentation in the java files.
@dr0i could you check. I opened a ticket for that: https://github.com/hbz/lobid-resources/issues/2023

For now I changed the indentation to tabss.

**Comment by TobiasNx** [comment]
@acka47 is this okay for you? We need an additional inCollection-statement for the digiBib Verbundkatalog since they now need NZ as well as CZ resources.

**Comment by TobiasNx** [comment]
@acka47 Is the way I modeled/mapped it okay for you?

```
  "inCollection" : [ {
    "id" : "https://nrw.digibib.net/search/hbzvk/",
    "label" : "DigiBib hbz Verbundkatalog",
    "type" : [ "Collection" ]
  }
```

**Comment by acka47** [comment]
> @acka47 Is the way I modeled/mapped it okay for you?

+1

**Review by acka47 — APPROVED** [review]
+1

**Pull request merged**